src/peer_transport/transport.ts

```diff
@@ -75,15 +76,7 @@ export class WebRTCTransport implements Transport, Startable {
     })
   }
 
-  /*
-   * dial connects to a remote via the circuit relay or any other protocol
-   * and proceeds to upgrade to a webrtc connection.
-   * multiaddr of the form: <multiaddr>/webrtc/p2p/<destination-peer>
-   * For a circuit relay, this will be of the form
-   * <relay address>/p2p/<relay-peer>/p2p-circuit/webrtc/p2p/<destination-peer>
-  */
-  async dial (ma: Multiaddr, options: DialOptions): Promise<Connection> {
-    log.trace('dialing address: ', ma)
+  private splitAddr (ma: Multiaddr): { baseAddr: Multiaddr, peerId: PeerId } {
     const addrs = ma.toString().split(`${TRANSPORT}/`)
     if (addrs.length !== 2) {
```

Comment:
not directly related but encountered in debugging, I think it's more intuitive to check if this is a `multiaddr` and then do the split.

```suggestion
if (!isMultiaddr(ma)) {
      throw new CodeError('invalid multiaddr', codes.ERR_INVALID_MULTIADDR)
}

const addrs = ma.toString().split(`${TRANSPORT}/`)
```

Reply:
This should be handled by the type system. We know this is a `Multiaddr` type in the param. When did you run into this not being a multiaddr?

The split and length check is to check if the `/webrtc/` part is in there. Really this should be interpreting the multiaddr from right to left, but the convention in JS has seemed to be to split on the string.

Reply:
> This should be handled by the type system. We know this is a Multiaddr type in the param. When did you run into this not being a multiaddr? 

Yes you're correct, silly oversight on my part 😓 

> he split and length check is to check if the /webrtc/ part is in there. Really this should be interpreting the multiaddr from right to left, but the convention in JS has seemed to be to split on the string.

Correct I would say this is a bit more intuitive, as someone reading this code I can't understand immediately why a post-split array with a length that's not equal to 2, is an invalid multiaddr.  Thanks for clarifying.